runtime/caml/tsan.h

```diff
@@ -93,10 +93,37 @@ CAMLextern void caml_tsan_exit_on_perform(uintnat pc, char* sp);
 CAMLextern void caml_tsan_entry_on_resume(uintnat pc, char* sp,
     struct stack_info const* stack);
 
+
+#if defined(WITH_THREAD_SANITIZER)
+
+// __tsan_func_exit can have either of the 2 signatures (#14082)
+#if defined(HAVE___TSAN_FUNC_EXIT_VOID_VOID_P)
 extern void __tsan_func_exit(void*);
+#elif defined(HAVE___TSAN_FUNC_EXIT_VOID_VOID)
+extern void __tsan_func_exit(void);
+#endif
+
 extern void __tsan_func_entry(void*);
 void __tsan_write8(void *location);
 
+Caml_inline void caml_tsan_func_exit(void) {
+#if defined(HAVE___TSAN_FUNC_EXIT_VOID_VOID_P)
+  __tsan_func_exit(NULL);
+#elif defined(HAVE___TSAN_FUNC_EXIT_VOID_VOID)
+  __tsan_func_exit();
+  #endif
+}
+
+Caml_inline void caml_tsan_func_entry(void *retaddr) {
+  __tsan_func_entry(retaddr);
+}
```

Comment:
The TSan CI now fails with non-trivial errors (stack entries disappearing and being replaced with multiples occurrences of `caml_tsan_entry_on_resume`). I suspect that you need to mark these two functions `CAMLno_tsan` to make sure that they don’t call `__tsan_func_{entry,exit}`.